src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -414,7 +414,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CompileWasmArgs Condition="'$(IlcLlvmExceptionHandlingModel)' == 'cpp'">$(CompileWasmArgs) -s DISABLE_EXCEPTION_CATCHING=0</CompileWasmArgs>
 
       <ScriptExt Condition="'$(OS)' == 'Windows_NT'">.bat</ScriptExt>
-      <WasmCompilerPath>&quot;$(EMSDK)/upstream/emscripten/emcc$(ScriptExt)&quot;</WasmCompilerPath>
+      <WasmCompilerPath Condition="'$(WasmCompilerPath)' == ''">&quot;$(EMSDK)/upstream/emscripten/emcc$(ScriptExt)&quot;</WasmCompilerPath>
```

Comment:
Is this override needed due to the layout difference between dotnet/emsdk and upstream emsdk?

Reply:
Exactly

Reply:
It is unfortunate that the layouts are different.

We will have to support both. It means creating an interface to the outside world that enables us to reason which layout is being used. There are two ways to do it:
1) Use a different environment variable (runtime build uses `EMSDK_PATH`) for the dotnet layout.
    - This would also need a precedence rule - which EMSDK wins if both are available.
2) Probe the layout, e. g. `UsingDotnetEmsdkLayout = Exists($(EMSDK)/upstream)`.

I see the upstream targets use `1`, and I prefer that approach too. I think the precedence should be that `EMSDK` wins, but with a warning (i. e. `if $(EMSDK) != '' && $(EMSDK_PATH) != '' warn("Both EMSDK and EMSDK_PATH set. Using EMSDK.")`.

Reply:
At the moment, I have it exactly like that. `EMSDK` wins and if `EMSDK` is empty, a `EmscriptenWorkloadPath` is used and `WasmCompilerPath` changed.

Reply:
What I am saying is that `WasmCompilerPath` should not be the interface to the SDK. `WasmCompilerPath` is an internal property (like `CompileWasmArgs`), just not named like that.

It's coincidental that we only use `emcc` from the SDK right now. We may need to use other tools (e. g. `llvm-ar`, to create static libraries, or `clang` for #2603). The logic that would select which layout to use should not need to be aware of what tools the targets happen to want to use.

What I am proposing is that we make `EMSDK_PATH` that interface property.

Reply:
Okay, makes sense